### PR TITLE
Fix duplicated tweets

### LINKED ISSUE #610: Check for duplicate tweets before posting
The automated twitter posts about pipeline releases are great, though sometimes releases can get double-postings (often because of something odd at a user-level with the release I think).

It might be nice to do a simple check before posting a tweet that there isn't already a tweet with identical text. This should avoid the simplest duplicate problems. Or perhaps instead of checking the entire tweet, just check the text at the start: `Pipeline release! nf-core/xxxx xxxx`

### PR DESCRIPTION
**Pull request by mashehu**
fixes #610 by fetching all nf-core tweets and searching for matching pipelinename+version.

<a href="https://gitpod.io/#https://github.com/nf-core/nf-co.re/pull/1229"></a>

### CODE REVIEW

File: update_pipeline_details.php [hunk 1]
```diff
@@ -242,11 +242,41 @@ function sort_datestamp($a, $b) {
...
+                        break;
+                    }
+                }
+
+                if (count($old_tweets) == 0 || $already_tweeted) {
+                    break;
+                }
+            }
+            if (!$already_tweeted) {
+                // $connection->post('statuses/update', ['status' => $tweet]);
```
Comment (ewels) [inline update_pipeline_details.php:277]: I think this needs to be uncommented?

File: update_pipeline_details.php [hunk 2]
```diff
@@ -242,11 +242,41 @@ function sort_datestamp($a, $b) {
...
+                $params = [
+                    'count' => 200,
+                    'page' => $page,
+                    'exclude_replies' => true,
+                    'exclude_retweets' => true,
+                ];
+                $old_tweets = $connection->get('statuses/user_timeline', $params);
+                // Check for substring of the new tweet in old tweets
+                foreach ($old_tweets as $old_tweet) {
+                    preg_match('/.*?\(/', $tweet, $match);
```
Comment (ewels) [inline update_pipeline_details.php:262]: This regex looks odd at a glance. Is it right?

**Review by ewels — CHANGES_REQUESTED** [review]

File: update_pipeline_details.php (hunk 1 above)
Comment (mashehu) [reply update_pipeline_details.php]: oops 🙃 I was very scared to actually send out a tweet while testing 😁

File: update_pipeline_details.php (hunk 2 above)
Comment (mashehu) [reply update_pipeline_details.php]: I made it stricter now (I think/hope).

File: update_pipeline_details.php (hunk 1 above)
Comment (ewels) [reply update_pipeline_details.php]: Better safe than sorry! 😅

File: update_pipeline_details.php (hunk 2 above)
Comment (ewels) [reply update_pipeline_details.php]: Looks good! I hadn't properly understood what it was doing but I get it now. I thought that you were using the regex to check for the name + version directly, had my noticed the second strpos check.

**Review by ewels — APPROVED** [review]

**Comment by ewels** [comment]
Now we wait! 🤞🏻

**Pull request merged**